File: src/SlimShader/Shader/SamplerMode.cs
using SlimShader.Util;

namespace SlimShader.Shader
{
	public enum SamplerMode
	{
		[Description("mode_default")]
		Default = 0,

		[Description("comparison")]
		Comparison = 1,

		[Description("mono")]
		Mono = 2
	}
}

File: src/cpp/Source/Chunks/Shex/SamplerMode.cpp
#include "PCH.h"
#include "SamplerMode.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(SamplerMode value)
{
	switch (value)
	{
	case SamplerMode::Default :
		return "mode_default";
	case SamplerMode::Comparison :
		return "comparison";
	case SamplerMode::Mono :
		return "mono";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}